### PR TITLE
Boot file selection UI control update + documentation on boot scripts feature.

### PR DESCRIPTION
**Pull request by ZiwKerman**
As discussed in issue #635 

Introduces dependency to KSPAPIExtensions.dll, uses UI_ChooseOption control to select boot script in editor.

For more information on KSPAPIExtensions see http://forum.kerbalspaceprogram.com/threads/81496-0-90-KSPAPIExtensions-V1-7-2-Utilities-for-shared-mod-use-16-Dec

I put the paragraph on boot scripts in `commands/files.rst` as it is related to files, but it could also be mentioned in `general/volumes.rst`

### CODE REVIEW

File: src/kOS/Module/kOSProcessor.cs [hunk 1]
```diff
@@ -132,6 +130,26 @@ public Screen.TermWindow GetWindow()
 
         public override void OnStart(StartState state)
         {
+			//Populate selector for boot scripts
```
Comment (erendrake) [inline src/kOS/Module/kOSProcessor.cs]: Would we want to do this if we are not in the editor? Also would you mind moving it into its own method?:

**Comment by erendrake** [comment]
Thank you very much for the PR :)

File: src/kOS/Module/kOSProcessor.cs (hunk 1 above)
Comment (ZiwKerman) [inline src/kOS/Module/kOSProcessor.cs]: I'll add a check for StartSate and move all the logic into a separate Method for a cleaner look.

**Comment by Dunbaratu** [comment]
To be honest I've never ever had the boot feature work and I never spoke up about how it's implemented because I don't have any examples of it working to compare against.  After this gets all merged in I might give it a try again.  Perhaps the problem I had is that since I dual boot back and forth between OSX and Windows 7, it might be the case that while under OSX I viewed the script folder with Mac's Finder program, which causes that hidden .DS_STORE file to appear (EVEN when you use Finder to view a non-UNIX filesystem where dotfiles aren't really hidden files like they are on UNIX filesystems, which is a infuriating), which apparently messed everything up.

**Comment by ZiwKerman** [comment]
I also have 2 macs - Mac Pro with Win10 installed for gaming and OSX for everything else, and a Macbook with OSX and Ubuntu installed. And boot script feature only worked for me on clean install of KSP + KOS + IR on my Mac Pro in OSX. All other installs ether refused to run boot scripts completely, or I wasn't able to get anything apart from default "boot.ks" to work, which was not enough for my purposes, as I wanted different boot scripts for several cores on my vessel (I want to automate RT network creation).

**Comment by erendrake** [comment]
@ZiwKerman have you looked at https://github.com/KSP-KOS/KOS/issues/618 ?

**Comment by ZiwKerman** [comment]
@erendrake I did, but it does not explain why my pure Win10 install wasn't working, so I decided to fix the problem myself :)

**Comment by erendrake** [comment]
One thing about this change @ZiwKerman. you sent us a PR against the master branch. We normally want requests to be for the develop branch.

this request also didnt add a reference to the csproj file for the new dependency 

Also.... tabs!!! 

this will not show as merged until i the next release because it is against master so ill just close it.

**Comment by ZiwKerman** [comment]
@erendrake 

Hmm, that was a mistake. I'm new to github, should've asked before. Also I could not compile develop branch maybe due to some changes by @Dunbaratu

**Comment by erendrake** [comment]
They were my changes that he merged in. I messed up and made some of the PRs dependent on one of the others. If you pull again it should be fixed.

**Comment by ZiwKerman** [comment]
And by "tabs!!!" you mean that I forgot to set autoconversion of tabs to spaces?

**Comment by ZiwKerman** [comment]
I pulled the update of develop branch, now it compiles. Should I resubmit a PR, for develop branch?

**Comment by erendrake** [comment]
Nope, i merged your changes across and resolved the issues. you are in :)

**Comment by ZiwKerman** [comment]
@pgodd 
At the first load on the launchpad after VAB/SPH selected boot file is copied into probe's local hard disk and named boot.ks

KOS.CPU.Boot(), which is called whenever probe comes into physics range from rails or when it is powered on from off state, checks whether there is a file "boot.ks" on starting volume and executes it.

You have to remember that boot file is only copied to probe's disk at the PRELAUNCH status, so any changes you make to the file afterwards would not be applied to probe's local copy, but you may manually update the file on probe's disk whenever you like.

I've checked 4 different scenarios:
 * switching to vessel from map view from outside of physics range - boot.ks is run
 * turning off the core and turning it back on - boot.ks is run
 * coming out of warp - boot is *not* run
 * switching between vessels in physics range - boot is *not* run

**Pull request closed without merging**